### DISCUSSION TITLE
Error 0x80370102

### DISCUSSION THREAD

**Discussion by arnavbhate** [microsoft/WSL#8364] [General]
### Version

Microsoft Windows [Version 10.0.19044.1645]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1

### Distro Version

Ubuntu 20.04

### Repro Steps

1. Install Virtual Machine Platform
2. Install wsl
3. Install Ubuntu with `wsl --install -d Ubuntu`
4. New Window pops up

### Expected Behavior

Ubuntu installs successfully and prompts for username and password

### Actual Behavior

```none
PS C:\Windows\System32> debian
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80370102
Error: 0x80370102 The virtual machine could not be started because a required feature is not installed.

```
Error 0x80370102

### Diagnostic Logs

[WslLogs-2022-05-04_10-01-09.zip](https://1drv.ms/u/s!AuOB0Lk4djILmdl8QGUhzio7DQoy5g?e=bZo0dx)
Excluding the output of the line
```powershell
get-acl "C:\ProgramData\Microsoft\Windows\WindowsApps" | Format-List > $folder/acl.txt
```
As I get an error here.

**Comment by arnavbhate** [comment]
I had opened #7879 in the past and following the steps suggested there I was able to fix the problem. However I am facing the problem again and those steps do not work.

**Reply by Biswa96** [reply]
Does `bcdedit` command as administrator show `hypervisorlaunchtype auto`?

**Reply by arnavbhate** [reply]
Yes

**Reply by njiangk** [reply]
Hey bro,see you again,and I recently ran into this same error,magical error🤣🤣

**Comment by ghost** [comment]
You may need to uninstall, reboot, reinstall, reboot, the "Virtual Machine Platform" under windows features.

![image](https://user-images.githubusercontent.com/102559858/166515017-47a2e21a-1d4b-4675-adeb-7ab0e66ea913.png)

**Reply by arnavbhate** [reply]
Already tried that

**Reply by ghost** [reply]
Without logs I can't do much.

**Reply by arnavbhate** [reply]
I can't collect logs because I get the error
```none
get-acl : Cannot find path 'C:\ProgramData\Microsoft\Windows\WindowsApps' because it does not exist.
At G:\collect-wsl-logs.ps1:31 char:1
+ get-acl "C:\ProgramData\Microsoft\Windows\WindowsApps" | Format-List  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (:) [Get-Acl], ItemNotFoundException
    + FullyQualifiedErrorId : GetAcl_PathNotFound_Exception,Microsoft.PowerShell.Commands.GetAclCommand
```

**Reply by arnavbhate** [reply]
@pmartincic Added logs excluding the error at
```powershell
get-acl "C:\ProgramData\Microsoft\Windows\WindowsApps" | Format-List > $folder/acl.txt
```

**Reply by ghost** [reply]
Using powershell, what do you get from the output of `Dism.exe /Online /Get-Features | Select-String -Pattern "irtual|yper" -Context 0,1` ?

**Reply by arnavbhate** [reply]
```none
> Feature Name : HypervisorPlatform
  State : Enabled
> Feature Name : VirtualMachinePlatform
  State : Enabled
> Feature Name : Microsoft-Hyper-V-All
  State : Enabled
> Feature Name : Microsoft-Hyper-V
  State : Enabled
> Feature Name : Microsoft-Hyper-V-Tools-All
  State : Enabled
> Feature Name : Microsoft-Hyper-V-Management-PowerShell
  State : Enabled
> Feature Name : Microsoft-Hyper-V-Hypervisor
  State : Enabled
> Feature Name : Microsoft-Hyper-V-Services
  State : Enabled
> Feature Name : Microsoft-Hyper-V-Management-Clients
  State : Enabled
  ```

**Reply by ghost** [reply]
So, the thing is: A variety of these features share common dependencies under the hood. Removing a subset of them would likely not cause the problematic components to be reinstalled. If you remove all of these, reboot, then install VirtualMachinePlatform, reboot, try again. If you still have problems after that then post back. At some point the components on your system ended up in a half installed inconsistent state. I'm working figure out how this system works and it's not at all part of our code.

**Reply by ghost** [reply]
And if you save off Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\Packages at each of those 3 points. I can't guarantee I'll be able to look further than that.

**Reply by arnavbhate** [reply]
[packages-1.reg](https://1drv.ms/u/s!AuOB0Lk4djILmdoxxsBGrJrvuxOObA?e=8fEcqa)

This is the state right now. You want me to remove Hyper-V, Virtual Machine Platform, Windows Hypervisor Platform and WSL and then send you another `reg` file? And then reinstall WSL and Virtual Machine Platform and send another?

**Reply by arnavbhate** [reply]
[Packages-2.reg](https://1drv.ms/u/s!AuOB0Lk4djILmdoy3IbKNv7Zmt0l6A?e=6Dejdz)
After uninstalling all of those

**Reply by arnavbhate** [reply]
[Packages-3.reg](https://1drv.ms/u/s!AuOB0Lk4djILmdozVNpheu24saEW8g?e=ZSj4pz)
After reinstalling WSL and Virtual Machine Platform

**Comment by Biswa96** [comment]
Would you like to try this command in a batch file?

```cmd
for %%X in (
HypervisorPlatform
Microsoft-Hyper-V
Microsoft-Windows-Subsystem-Linux
VirtualMachinePlatform ) do (
    DISM.exe /Online /Enable-Feature /All /FeatureName:%%X /NoRestart
)
```

Then restart and try to use WSL2.

**Reply by arnavbhate** [reply]
Didn't work

**Comment by njiangk** [comment]
I found that when I upgraded windows 11 to 22h2 this problem was solved

**Reply by arnavbhate** [reply]
I am on Windows 10 and my processor is not compatible with Windows 11

**Reply by Biswa96** [reply]
Sometimes hard errors can be solved just by reinstalling Windows OS. It doesn't matter if it's Win10 or 11. But that option isn't convenient for all users.

**Reply by arnavbhate** [reply]
Reinstalling is my last option if all else fails
